src/App.tsx

```diff
@@ -1,12 +1,92 @@
 import React from 'react';
 import './App.scss';
+import classNames from 'classnames';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodList } from './GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods: Good[] | null;
+  sortBy: string;
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: null,
+    sortBy: 'all',
+  };
+
+  showAll = () => {
```

Comment:
Instead of these methods, you could create a method that would accept a function passed from .api and return the relevant list of goods.